File: 2439-minimize-maximum-of-array/2439-minimize-maximum-of-array.java
class Solution {
    public int minimizeArrayValue(int[] nums) {
        long sum=0,mn=0;
        for(int i=0;i<nums.length;i++){
            sum+=nums[i];
            mn=Math.max(mn,(sum+i)/(i+1));
        }
        return (int)mn;
    }
}

File: 2439-minimize-maximum-of-array/2439-minimize-maximum-of-array.cpp
class Solution {
public:
    int minimizeArrayValue(vector<int>& nums) {
        long long sum=0,mn=0;
        for(int i=0;i<nums.size();i++){
            sum+=nums[i];
            mn=max(mn,(sum+i)/(i+1));
        }
        return mn;
    }
};